storage/txcache/txCacheFailsafe.go

```diff
@@ -0,0 +1,140 @@
+package txcache
+
+import (
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+var _ storage.Cacher = (*txCacheFailsafeDecorator)(nil)
+var _ txCache = (*txCacheFailsafeDecorator)(nil)
+
+// txCacheFailsafeDecorator is a decorator over a cache, a decorator which wraps some operations in try / catch constructs
+type txCacheFailsafeDecorator struct {
+	backingCache txCache
+}
+
+func newTxCacheFailsafe(cache txCache) txCache {
+	return &txCacheFailsafeDecorator{
+		backingCache: cache,
+	}
+}
+
+// AddTx delegates to backing cache, with try / catch
+func (decorator *txCacheFailsafeDecorator) AddTx(tx *WrappedTransaction) (ok bool, added bool) {
+	core.TryCatch(func() {
+		ok, added = decorator.backingCache.AddTx(tx)
+	}, decorator.catchAndLog, "AddTx")
+
+	return
+}
+
+// GetByTxHash delegates to backing cache
+func (decorator *txCacheFailsafeDecorator) GetByTxHash(txHash []byte) (*WrappedTransaction, bool) {
+	return decorator.backingCache.GetByTxHash(txHash)
+}
+
+// SelectTransactions deledates to backing cache, with try / cache
+func (decorator *txCacheFailsafeDecorator) SelectTransactions(numRequested int, batchSizePerSender int) []*WrappedTransaction {
+	selection := make([]*WrappedTransaction, 0)
+
+	core.TryCatch(func() {
+		selection = decorator.backingCache.SelectTransactions(numRequested, batchSizePerSender)
```

Comment:
also this? Why this is allowed to panic?

Reply:
Removed, see above.